libbzip2-rs-sys/src/bzlib.rs

```diff
@@ -734,27 +734,31 @@ pub(crate) fn BZ2_bzCompressInitHelp(
         }
     };
 
-    unsafe {
-        (*s).blockNo = 0;
-        (*s).state = State::Output;
-        (*s).mode = Mode::Running;
-        (*s).combinedCRC = 0;
-        (*s).blockSize100k = blockSize100k;
-        (*s).nblockMAX = 100000 * blockSize100k - 19;
-        (*s).verbosity = verbosity;
-        (*s).workFactor = workFactor;
-    }
-
     strm.state = s;
 
+    // safety: the EState has now been sufficiently initialized; the allocator zeroes the memory,
+    // and the only fields where zero is not a valid value are the arrays that were just set
+    //
+    // note in particular that if the discriminant of the first variant of an enum is unspecified,
+    // then it is set to zero.
```

Comment:
This is not guaranteed unless you have a `#[repr(C)]` or `#[repr(uN)]`/`#[repr(iN)]` enum.

Reply:
well, it's what the reference says

https://doc.rust-lang.org/reference/items/enumerations.html#implicit-discriminants

> If a discriminant for a variant is not specified, then it is set to one higher than the discriminant of the previous variant in the declaration. If the discriminant of the first variant in the declaration is unspecified, then it is set to zero.

is that inaccurate?

Reply:
The discriminant (the value you get when casting the enum to an integer) is guaranteed to be zero, but the tag (the actual encoding of the discriminant when not doing niche-filling) is not guaranteed to be zero afaik.

Reply:
well we'll just make them repr(u8) then (because these are fields in a `repr(C)` struct I think that changes nothing)